packages/synapse-react-client/src/utils/AppUtils/AppUtils.ts

```diff
@@ -33,6 +36,16 @@ export function storeRedirectURLForOneSageLoginAndGotoURL(href: string) {
 }
 
 export function processRedirectURLInOneSage() {
+  // PORTALS-3299 : Indicate that we have completed the login workflow (cookie expires in a minute) to break out of a cycle
+  const expireDate = new Date()
+  expireDate.setMinutes(expireDate.getMinutes() + 1)
+  const hostname = window.location.hostname.toLowerCase()
+  cookies.set(ACCOUNT_SITE_PROMPTED_FOR_LOGIN_COOKIE_KEY, 'true', {
+    path: '/',
+    expires: expireDate,
+    domain: hostname.endsWith('.synapse.org') ? 'synapse.org' : undefined,
```

Comment:
This won't work on dev (which is OK for now, since we don't have a dev OAuth app or accounts site)